AppLovin MAX Demo App - Swift/AppLovin MAX Demo App - Swift/MAX/Native Ads/UI/ALMAXSwiftUITemplateNativeAdView.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  ALMAXSwiftUITemplateNativeAdView.swift
+//  AppLovin MAX Demo App - Swift
+//
+//  Created by Matthew Nguyen on 8/1/23.
+//  Copyright © 2023 AppLovin. All rights reserved.
+//
+
+import SwiftUI
+import Adjust
+import AppLovinSDK
+
+@available(iOS 13.0, *)
+struct ALMAXSwiftUITemplateNativeAdView: View
+{
+    @ObservedObject private var viewModel = ALMAXSwiftUITemplateNativeAdViewModel(adUnitId: "YOUR_AD_UNIT_ID")
+    
+    var body: some View {
+        
+        VStack {
+            MANativeTemplateAdViewSwiftUIWrapper(triggerShowAd: $viewModel.triggerShowAd,
+                                                 nativeAdLoader: viewModel.adLoader,
+                                                 containerView: viewModel.containerView,
+                                                 didLoadNativeAd: viewModel.didLoadNativeAd(_:for:),
+                                                 didFailToLoadNativeAd: viewModel.didFailToLoadNativeAd(forAdUnitIdentifier:withError:),
+                                                 didClickNativeAd: viewModel.didClickNativeAd(_:),
+                                                 didExpireNativeAd: viewModel.didExpireNativeAd(_:),
+                                                 didPayRevenue: viewModel.didPayRevenue(for:))
+                .frame(width: 300, height: 250)
+            
+            callbacksTable
+                .frame(maxHeight: .infinity)
+            Button {
+                viewModel.triggerShowAd = true
+            } label: {
+                Text("Show")
+            }
+        }
+    }
+    
+    var callbacksTable: some View {
+        List(viewModel.callbacks) {
+            Text($0.callback)
+        }
+    }
+}
+
+@available(iOS 13.0, *)
+class ALMAXSwiftUITemplateNativeAdViewModel: NSObject, ObservableObject
+{
+    @Published fileprivate var callbacks: [CallbackTableItem] = []
+    @Published var triggerShowAd: Bool = false
+    
+    let adUnitIdentifier: String
+    
+    let adLoader: NativeSwiftUIAdLoader
```

Comment:
I noticed we keep a reference to the `adLoader` in the view model, but don't use it in there (rather using it mainly in the wrapper). Architecture could be fine, though I'd double check if we need those two references or we could just initialize it in the wrapper.

Reply:
Apologies - I'm a bit unclear on this could you please clarify to see if I'm understanding this feedback correctly?

The view model initializes an `adLoader` of `ALMAXNativeSwiftUIAdLoader` type (previously called `NativeSwiftUIAdLoader`) which is a wrapper for the SDK's `MANativeAdLoader`. The view model uses the `adLoader` in its [`didLoadNativeAd` method](https://github.com/AppLovin/AppLovin-MAX-SDK-iOS/blob/069a4fae1e3589d237ed653a6d65ec010aaf0229/AppLovin%20MAX%20Demo%20App%20-%20Swift/AppLovin%20MAX%20Demo%20App%20-%20Swift/MAX/Native%20Ads/UI/ALMAXSwiftUITemplateNativeAdView.swift#L84) where the ad loader manages the loaded `nativeAdView` into its own `containerView`.  This view model gets initialized in `ALMAXSwiftUITemplateNativeAdView` and the `ALMAXNativeSwiftUIAdLoader` that got initialized from the view model is passed into the `ALMAXNativeTemplateAdViewSwiftUIWrapper` so the [ad loader can set it's delegates and show ads whenever triggered to. ](https://github.com/AppLovin/AppLovin-MAX-SDK-iOS/blob/069a4fae1e3589d237ed653a6d65ec010aaf0229/AppLovin%20MAX%20Demo%20App%20-%20Swift/AppLovin%20MAX%20Demo%20App%20-%20Swift/Base%20Classes/MANativeTemplateAdViewSwiftUIWrapper.swift#L29)

I could technically have the one reference of the `adLoader` inside just `ALMAXNativeTemplateAdViewSwiftUIWrapper` instead of the view model by doing the adView management inside there, but I wanted to follow the coding conventions that already exist from the `MAAdViewSwiftUIWrapper` where the [coordinator class](https://github.com/AppLovin/AppLovin-MAX-SDK-iOS/blob/069a4fae1e3589d237ed653a6d65ec010aaf0229/AppLovin%20MAX%20Demo%20App%20-%20Swift/AppLovin%20MAX%20Demo%20App%20-%20Swift/Base%20Classes/MAAdViewSwiftUIWrapper.swift#L63) of the delegates simply call the parent object's delegate methods [which are defined in the view model.](https://github.com/AppLovin/AppLovin-MAX-SDK-iOS/blob/069a4fae1e3589d237ed653a6d65ec010aaf0229/AppLovin%20MAX%20Demo%20App%20-%20Swift/AppLovin%20MAX%20Demo%20App%20-%20Swift/MAX/Banners/UI/ALMAXSwiftUIBannerAdView.swift#L61) I'm unsure if this is what you were looking for.

Let me know if my explanation of the current architecture makes sense - I could be missing the big picture behind your feedback.

(Upon me trying to understand this feedback, I began to realize my naming of `adLoader` can be inconsistent across the different wrappers and view model so I will begin making them more consistent to avoid any more confusion.)

Reply:
Ah never mind, I totally neglected the extensions below it being for the view model. This looks perfectly fine to me then!